dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h

```diff
@@ -102,19 +102,20 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
                                        TensorShape<1>{300*1024});
       hw_decoder_images_staging_.Resize(shapes, TypeInfo::Create<uint8_t>());
 #if defined(NVJPEG_PREALLOCATE_API)
-      // TODO(awolant): How to expose chroma subsampling to the user?
-      if (spec.GetArgument<int>("preallocate_width_hint") > 0 &&
-          spec.GetArgument<int>("preallocate_height_hint") > 0) {
-        LOG_LINE << "Using NVJPEG_PREALLOCATE_API" << std::endl;
-        NVJPEG_CALL(nvjpegDecodeBatchedPreAllocate(
-          handle_,
-          state_hw_batched_,
-          CalcHwDecoderBatchSize(hw_decoder_load_, max_batch_size_),
-          spec.GetArgument<int>("preallocate_width_hint"),
-          spec.GetArgument<int>("preallocate_height_hint"),
-          NVJPEG_CSS_444,
-          NVJPEG_OUTPUT_RGB));
-      }
+      // call nvjpegDecodeBatchedPreAllocate to use memory pool for HW decoder even if hint is 0
+      auto preallocate_width_hint = spec.GetArgument<int>("preallocate_width_hint");
+      auto preallocate_height_hint = spec.GetArgument<int>("preallocate_height_hint");
+      preallocate_width_hint = preallocate_width_hint < 0 ? 0 : preallocate_width_hint;
+      preallocate_height_hint = preallocate_height_hint < 0 ? 0 : preallocate_height_hint;
```

Comment:
I know it works, but I'm not sure why you used `0 < ` instead of `HasArgument` here? Default value for those is 0, so the condition is never true, I think.
Maybe these lines are not needed after all? Or, if you don't want to rely on default value, making the condition use `HasArgument` would better reflect intent behind this? Just a thought, I'm fine either way.

Reply:
Just in case someone provides a negative value.
@mzient what do you think?

Reply:
Well, if someone provides a negative value, I think we should protest. We already have a special value in Python (None) that user can pass to say "use the default", so using negative values to state the same seems redundant.

Reply:
Added check for negative values, default 0 is fine for us.